### PR TITLE
Doc/cyclic symmetry

### PR DESCRIPTION
**Pull request by roosre**
Add an example to demonstrate that pyDPF Composites can be used for results of cyclic symmetry analyses as well.

While playing with the cyclic symmetry model and going through the DPF Core examples, I figured out that the RST streams provider is often used. Therefore, I thought it makes sense to expose it through a method. Let me know if you have objections.

### CODE REVIEW

File: examples/014_cyclic_symmetry_example.py [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+"""
+.. _cyclic_symmetry_example:
+
+Cyclic symmetry
+---------------
+
+This example shows how to post-process a cyclic symmetry analysis.
```
Comment (nshum4) [inline examples/014_cyclic_symmetry_example.py:29]:
```suggestion
This example shows how to postprocess a cyclic symmetry analysis.
```

File: examples/014_cyclic_symmetry_example.py [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+The initial (original) sector can be post-processed with the same tools
```
Comment (nshum4) [inline examples/014_cyclic_symmetry_example.py:30]:
```suggestion
The initial (original) sector can be postprocessed with the same tools
```

File: examples/014_cyclic_symmetry_example.py [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+as a standard analysis. This is demonstrated by running a failure analysis,
```
Comment (nshum4) [inline examples/014_cyclic_symmetry_example.py:31]: I think this should be clarified: ... with the same tools as a standard analysis [of what?].

File: examples/014_cyclic_symmetry_example.py [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+extracting ply-wise stresses and the implementation of a custom
```
Comment (nshum4) [inline examples/014_cyclic_symmetry_example.py:32]: Question about this sentence (including the one revised below): Is the failure analysis extracting both ply-wise stresses and the implementation, or extracting only the ply-wise stresses along with implementing a custom failure criterion?

```suggestion
 The postprocessing workflow is demonstrated by running a failure analysis that extracts ply-wise stresses and the implementation of a custom failure criterion.
```

File: examples/014_cyclic_symmetry_example.py [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+failure criterion.
+
+The post-processing of the expanded sectors is not yet supported.
```
Comment (nshum4) [inline examples/014_cyclic_symmetry_example.py:35]:
```suggestion
The postprocessing of expanded sectors is not yet supported.
```

File: examples/014_cyclic_symmetry_example.py [hunk 6]
```diff
@@ -0,0 +1,160 @@
...
+"""
+
+# %%
+# Set up analysis
+# ~~~~~~~~~~~~~~~
+# Setting up the analysis consists of loading the required modules, connecting to the
+# DPF server, and retrieving the example files.
+#
+# Load Ansys libraries and helper functions
```
Comment (nshum4) [inline examples/014_cyclic_symmetry_example.py:44]:
```suggestion
# Load Ansys libraries and helper functions.
```

File: examples/014_cyclic_symmetry_example.py [hunk 7]
```diff
@@ -0,0 +1,160 @@
...
+from ansys.dpf.composites.select_indices import get_selected_indices
+from ansys.dpf.composites.server_helpers import connect_to_or_start_server
+
+# %%
+# Start a DPF server and copy the example files into the current working directory.
+server = connect_to_or_start_server()
+composite_files = get_continuous_fiber_example_files(server, "cyclic_symmetry")
+
+# %%
+# Create a composite model
```
Comment (nshum4) [inline examples/014_cyclic_symmetry_example.py:63]:
```suggestion
# Create a composite model.
```

File: examples/014_cyclic_symmetry_example.py [hunk 8]
```diff
@@ -0,0 +1,160 @@
...
+composite_model = CompositeModel(composite_files, server)
+
+# %%
+# Evaluate a combined failure criterion
```
Comment (nshum4) [inline examples/014_cyclic_symmetry_example.py:67]:
```suggestion
# Evaluate a combined failure criterion.
```

File: examples/014_cyclic_symmetry_example.py [hunk 9]
```diff
@@ -0,0 +1,160 @@
...
+combined_failure_criterion = CombinedFailureCriterion(failure_criteria=[MaxStressCriterion()])
+failure_result = composite_model.evaluate_failure_criteria(combined_failure_criterion)
+
+# %%
+# Plot the failure results
```
Comment (nshum4) [inline examples/014_cyclic_symmetry_example.py:72]:
```suggestion
# Plot the failure results.
```

File: examples/014_cyclic_symmetry_example.py [hunk 10]
```diff
@@ -0,0 +1,160 @@
...
+irf_field = failure_result.get_field({"failure_label": FailureOutput.FAILURE_VALUE})
+irf_field.plot()
+
+# %%
+# Plot ply-wise stresses
+# ~~~~~~~~~~~~~~~~~~~~~~
+# All functions in PyDPF Composites can be used to
+# post-process the original (first) sector.
```
Comment (nshum4) [inline examples/014_cyclic_symmetry_example.py:80]:
```suggestion
# postprocess the initial (original) sector.
```

File: examples/014_cyclic_symmetry_example.py [hunk 11]
```diff
@@ -0,0 +1,160 @@
...
+    spot_reduction_strategy=SpotReductionStrategy.MAX,
+    requested_location=dpf.locations.elemental,
+)
+composite_model.get_mesh().plot(elemental_values)
+
+# %%
+# Custom failure criterion
+# ~~~~~~~~~~~~~~~~~~~~~~~~
+# The following code block shows how to implement a custom failure criterion.
+# It computes the inverse reserve factor for each element with respect to
```
Comment (nshum4) [inline examples/014_cyclic_symmetry_example.py:107]:
```suggestion
# It computes the inverse reserve factor (IRF) for each element with respect to
```

File: examples/014_cyclic_symmetry_example.py [hunk 12]
```diff
@@ -0,0 +1,160 @@
...
+# fiber failure. The criterion distinguishes between tension and compression.
+
+# Prepare dict with the material properties
```
Comment (nshum4) [inline examples/014_cyclic_symmetry_example.py:110]:
```suggestion
# Prepare a dictionary (dict) with the material properties.
```

File: examples/014_cyclic_symmetry_example.py [hunk 13]
```diff
@@ -0,0 +1,160 @@
...
+property_xt = MaterialProperty.Stress_Limits_Xt
+property_xc = MaterialProperty.Stress_Limits_Xc
+property_dict = composite_model.get_constant_property_dict([property_xt, property_xc])
+
+result_field = dpf.field.Field(location=dpf.locations.elemental, nature=dpf.natures.scalar)
+with result_field.as_local_field() as local_result_field:
+    # process only the layered elements
```
Comment (nshum4) [inline examples/014_cyclic_symmetry_example.py:117]:
```suggestion
    # Process only the layered elements
```

File: examples/014_cyclic_symmetry_example.py [hunk 14]
```diff
@@ -0,0 +1,160 @@
...
+                element_irf_max = max(min_s11 / xc, element_irf_max)
+
+        local_result_field.append([element_irf_max], element_id)
+
+composite_model.get_mesh().plot(result_field)
+
+# %%
+# Plot deformations on the expanded model
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+# For the sack of completeness, it is shown how to expand the
```
Comment (nshum4) [inline examples/014_cyclic_symmetry_example.py:142]:
```suggestion
# You can expand the deformations of the cyclic symmetry model as shown below. The same expansion is possible for strains and stresses. For more information, see `Ansys DPF`_.
```

File: examples/014_cyclic_symmetry_example.py [hunk 15]
```diff
@@ -0,0 +1,160 @@
...
+# deformations of the cyclic symmetry model. The same can be done
+# for strains and stresses. See `Ansys DPF`_ for more details.
+#
+# .. _Ansys DPF: https://dpf.docs.pyansys.com/version/stable/
+
+# Get the displacements and expand them
```
Comment (nshum4) [inline examples/014_cyclic_symmetry_example.py:148]:
```suggestion
# Get the displacements and expand them.
```

**Review by nshum4 — CHANGES_REQUESTED** [review]
Hi René! I finished reviewing example 14. I committed my changes & left a few comments with questions.

Can you let me know if there is anything else I can do to help?

File: examples/014_cyclic_symmetry_example.py (hunk 12 above)
Comment (nshum4) [reply examples/014_cyclic_symmetry_example.py:110]: @roosre Is elaborating on dict benefical? Or do the users know what it means already?

File: examples/014_cyclic_symmetry_example.py [hunk 16]
```diff
@@ -0,0 +1,160 @@
...
+                element_irf_max = max(min_s11 / xc, element_irf_max)
+
+        local_result_field.append([element_irf_max], element_id)
+
+composite_model.get_mesh().plot(result_field)
+
+# %%
+# Plot deformations on the expanded model
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+# You can expand the deformations of the cyclic symmetry model as shown below. The same expansion is possible for strains and stresses. For more information, see `Ansys DPF`_.
```
Comment (nshum4) [inline examples/014_cyclic_symmetry_example.py:142]:
```suggestion
# You can expand the deformations of the cyclic symmetry model as shown below. 
The same expansion is possible for strains and stresses. For more information, see `Ansys DPF`_.
```

File: examples/014_cyclic_symmetry_example.py [hunk 17]
```diff
@@ -0,0 +1,161 @@
...
+
+        local_result_field.append([element_irf_max], element_id)
+
+composite_model.get_mesh().plot(result_field)
+
+# %%
+# Plot deformations on the expanded model
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+# You can expand the deformations of the cyclic symmetry model as shown below. 
+The same expansion is possible for strains and stresses. For more information, see `Ansys DPF`_.
```
Comment (nshum4) [inline examples/014_cyclic_symmetry_example.py:143]:
```suggestion
# The same expansion is possible for strains and stresses. For more information, see `Ansys DPF`_.
```

File: examples/014_cyclic_symmetry_example.py [hunk 18]
```diff
@@ -0,0 +1,160 @@
...
+        local_result_field.append([element_irf_max], element_id)
+
+composite_model.get_mesh().plot(result_field)
+
+# %%
+# Plot deformations on the expanded model
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+# You can expand the deformations of the cyclic symmetry model as shown below. 
+# The same expansion is possible for strains and stresses. For more information, see `Ansys DPF`_.
+# deformations of the cyclic symmetry model. The same can be done
```
Comment (nshum4) [inline examples/014_cyclic_symmetry_example.py:144]:
```suggestion
```

File: examples/014_cyclic_symmetry_example.py
Comment (nshum4) [inline examples/014_cyclic_symmetry_example.py:1]: @roosre So, I committed a change to this section in GitHub & broke one of the checks. I don't know how to fix it. Any ideas?

File: examples/014_cyclic_symmetry_example.py (hunk 12 above)
Comment (roosre) [reply examples/014_cyclic_symmetry_example.py:110]: Hi @nshum4 , a python user knows what a dict is. It's a native python class (object) and widely used. Thanks, René

File: examples/014_cyclic_symmetry_example.py
Comment (roosre) [reply examples/014_cyclic_symmetry_example.py:1]: Hi @nshum4 , it's fixed. It was just a style check (unneeded space at the end of a line). Did you also review example 013_thermal_example.py?

Comment (nshum4) [reply examples/014_cyclic_symmetry_example.py:1]: Oh, good. Thank you! & I didn't review example 13 yet. I'll get to it either today or tomorrow.

File: examples/014_cyclic_symmetry_example.py (hunk 12 above)
Comment (nshum4) [reply examples/014_cyclic_symmetry_example.py:110]:
```suggestion
# Prepare dict with the material properties.
```

File: examples/014_cyclic_symmetry_example.py (hunk 3 above)
Comment (roosre) [reply examples/014_cyclic_symmetry_example.py:31]: same tools stands for features and functions of this python module.
standard analysis stands for eg. static structural analysis.

Comment (nshum4) [reply examples/014_cyclic_symmetry_example.py:31]: Got it. Thank you for helping me understand this. I'll resolve as it is!

**Review by nshum4 — APPROVED** [review]

File: examples/014_cyclic_symmetry_example.py (hunk 4 above)
Comment (roosre) [reply examples/014_cyclic_symmetry_example.py:32]: Hi @nshum4 , these are 3 independent examples: 
- failure analysis
- extracting ply-wise stresses
- implementation of a custom failure criterion

I agree, it's not clear how it is written currently.

File: examples/014_cyclic_symmetry_example.py [hunk 19]
```diff
@@ -0,0 +1,160 @@
...
+.. _cyclic_symmetry_example:
+
+Cyclic symmetry
+---------------
+
+This example shows how to post-process a cyclic symmetry analysis.
+The initial (original) sector can be post-processed with the same tools
+as a standard analysis. This is demonstrated by running a failure analysis,
+extracting ply-wise stresses and the implementation of a custom
+failure criterion.
```
Comment (nshum4) [inline examples/014_cyclic_symmetry_example.py:33]:
```suggestion
as a standard analysis. The postprocessing workflow is demonstrated by
running a failure analysis, extracting ply-wise stresses, and implementing
a custom failure criterion.
```

Comment (nshum4) [reply examples/014_cyclic_symmetry_example.py:33]: @roosre Thank you for the feedback! If this sentence is clear, can you commit it?

File: examples/014_cyclic_symmetry_example.py [hunk 20]
```diff
@@ -0,0 +1,159 @@
...
+Cyclic symmetry
+---------------
+
+This example shows how to postprocess a cyclic symmetry analysis.
+The initial (original) sector can be postprocessed with the same tools
+as a standard analysis. The postprocessing workflow is demonstrated by
+running a failure analysis, extracting ply-wise stresses, and implementing
+a custom failure criterion.
+
+The postprocessing of expanded sectors is not yet supported.
```
Comment (greschd) [inline examples/014_cyclic_symmetry_example.py:35]: Should this go into a "Limitations" section?

**Review by greschd — APPROVED** [review]

**Pull request merged**